judge-server/judge/compiler.py

```
@@ -5,49 +5,78 @@
-        "--tmpfs", "/tmp:rw,noexec,nosuid,size=32m",
+
+    docker_cmd = _build_docker_command(work_dir, DOCKER_IMAGES["JAVA"])
+    docker_cmd.extend(["javac", "-encoding", "UTF-8", "-cp", ".", "Main.java"])
+
+    return _run_compilation(docker_cmd, "JAVA")
+
+def compile_python(source_code_path) -> int:
+    work_dir = os.path.dirname(source_code_path)
+
+    docker_cmd = _build_docker_command(work_dir, DOCKER_IMAGES["PYTHON"])
+    docker_cmd.extend(["python", "-W", "ignore", "-c", f"import py_compile; py_compile.compile(r'Main.py')"])
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)